### PR TITLE
Fix BPF compile on minikube

### LINKED ISSUE #503: KubeArmor crashes on minikube because BPF code can't compile
## Bug Report
**General Information**

- Environment description (GKE, VM-Kubeadm, vagrant-dev-env, minikube, microk8s, ...)
```
minikube version: v1.24.0
commit: 76b94fb3c4e8ac5062daf70d60cf03ddcc0a741b
```
- Kernel version (run `uname -a`)
```
Linux minikube 4.19.94 #2 SMP Mon Mar 30 23:26:19 UTC 2020 x86_64 GNU/Linux
```
- Kernel configuration (run `zcat /proc/config.gz` or `cat /lib/modules/$(uname -r)/build/.config`)
```
...
CONFIG_HAVE_KPROBES_ON_FTRACE=y
CONFIG_HAVE_DYNAMIC_FTRACE=y
CONFIG_HAVE_DYNAMIC_FTRACE_WITH_REGS=y
CONFIG_HAVE_FTRACE_MCOUNT_RECORD=y
CONFIG_FTRACE=y
# CONFIG_FTRACE_SYSCALLS is not set
```
- Orchestration system version in use (e.g. `kubectl version`, ...)
All
- Link to relevant artifacts (policies, deployments scripts, ...)
[KubeArmor minikube deployment](https://github.com/kubearmor/KubeArmor/blob/main/deployments/minikube/kubearmor.yaml)

**To Reproduce**

1. Setup minikube cluster using [this](https://github.com/kubearmor/KubeArmor/tree/main/contribution/minikube) guide.
2. Apply kubearmor minikube deployment yaml. KubeArmor will keep crashing.
```
NAMESPACE     NAME                                             READY   STATUS      RESTARTS        AGE
kube-system   coredns-78fcd69978-wf9tn                         1/1     Running     0               34m
kube-system   etcd-minikube                                    1/1     Running     0               34m
kube-system   kube-apiserver-minikube                          1/1     Running     0               34m
kube-system   kube-controller-manager-minikube                 1/1     Running     0               34m
kube-system   kube-flannel-ds-amd64-j4ng2                      1/1     Running     0               34m
kube-system   kube-proxy-klv4g                                 1/1     Running     0               34m
kube-system   kube-scheduler-minikube                          1/1     Running     0               34m
kube-system   kubearmor-host-policy-manager-69cfc96948-fwcqt   2/2     Running     0               17m
kube-system   kubearmor-policy-manager-986bd8dbc-fsdv8         2/2     Running     0               17m
kube-system   kubearmor-ptmp9                                  0/1     Completed   7 (5m53s ago)   17m
kube-system   kubearmor-relay-645667c695-smc5q                 0/1     Completed   6 (5m52s ago)   17m
kube-system   storage-provisioner                              1/1     Running     0               34m
```
3. Check logs using `kubectl logs -n kube-system kubearmor-XYZW`.
```
2021-11-20 11:29:08.154684	INFO	Initialized Kubernetes client
2021-11-20 11:29:08.154759	INFO	Started to monitor node events
2021-11-20 11:29:09.155230	INFO	Initialized KubeArmor Logger
2021-11-20 11:29:09.155283	INFO	Started to serve gRPC-based log feeds
2021-11-20 11:29:09.156135	INFO	Initializing an eBPF program
/virtual/main.c:1030:1: warning: declaration of 'struct tracepoint__syscalls__sys_exit_openat' will not be visible outside of this function [-Wvisibility]
TRACEPOINT_PROBE(syscalls, sys_exit_openat)
^
/virtual/include/bcc/helpers.h:1211:46: note: expanded from macro 'TRACEPOINT_PROBE'
int tracepoint__##category##__##event(struct tracepoint__##category##__##event *args)
                                             ^
<scratch space>:106:1: note: expanded from here
tracepoint__syscalls__sys_exit_openat
^
/virtual/main.c:1051:26: error: incomplete definition of type 'struct tracepoint__syscalls__sys_exit_openat'
    context.retval = args->ret;
                     ~~~~^
/virtual/main.c:1030:1: note: forward declaration of 'struct tracepoint__syscalls__sys_exit_openat'
TRACEPOINT_PROBE(syscalls, sys_exit_openat)
^
/virtual/include/bcc/helpers.h:1211:46: note: expanded from macro 'TRACEPOINT_PROBE'
int tracepoint__##category##__##event(struct tracepoint__##category##__##event *args)
                                             ^
<scratch space>:106:1: note: expanded from here
tracepoint__syscalls__sys_exit_openat
^
1 warning and 1 error generated.
2021-11-20 11:29:10.470886	ERROR	bpf module is nil
github.com/kubearmor/KubeArmor/KubeArmor/log.Err
	/usr/src/KubeArmor/KubeArmor/log/logger.go:97
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).InitSystemMonitor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:234
github.com/kubearmor/KubeArmor/KubeArmor/core.KubeArmor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:410
main.main
	/usr/src/KubeArmor/KubeArmor/main.go:50
runtime.main
	/usr/local/go/src/runtime/proc.go:204
2021-11-20 11:29:10.470966	ERROR	Failed to initialize KubeArmor Monitor
github.com/kubearmor/KubeArmor/KubeArmor/log.Err
	/usr/src/KubeArmor/KubeArmor/log/logger.go:97
github.com/kubearmor/KubeArmor/KubeArmor/feeder.(*Feeder).Err
	/usr/src/KubeArmor/KubeArmor/feeder/feeder.go:486
github.com/kubearmor/KubeArmor/KubeArmor/core.KubeArmor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:411
main.main
	/usr/src/KubeArmor/KubeArmor/main.go:50
runtime.main
	/usr/local/go/src/runtime/proc.go:204
2021-11-20 11:29:10.471005	INFO	Stopped KubeArmor Monitor
2021-11-20 11:29:10.471011	INFO	Terminated KubeArmor
2021-11-20 11:29:12.471670	INFO	Terminated the gRPC service
2021-11-20 11:29:12.471809	INFO	Stopped KubeArmor Logger
2021-11-20 11:29:12.471835	INFO	Waiting for routine terminations
```

**Expected behavior**

KubeArmor should be able to run on minikube. 

**Screenshots and other observations**
1. Tracepoint event directory for syscalls is missing on minikube image.
```
# cd /sys/kernel/debug/tracing/events/syscalls
-sh: cd: /sys/kernel/debug/tracing/events/syscalls: No such file or directory
```
2. `CONFIG_FTRACE_SYSCALLS` is not set which is required for `sys_exit_openat` tracepoint. We can also notice that ftrace is missing in available tracers.
```
# cat /sys/kernel/debug/tracing/available_tracers 
blk nop
```

### PR DESCRIPTION
**Pull request by h3llix**
Fixes: #503 

Signed-off-by: Gaurav <h3llix.pvt@gmail.com>

### CODE REVIEW

File: contribution/minikube/start_minikube.sh [hunk 1]
```diff
@@ -3,9 +3,11 @@
 # Copyright 2021 Authors of KubeArmor
 
 # Refer to https://minikube.sigs.k8s.io/docs/tutorials/ebpf_tools_in_minikube
+# download minikube iso
+wget -O minikube.iso https://github.com/kubearmor/kastore/blob/main/iso/minikube/minikube.iso?raw=true
```
Comment (nam-jaehyun) [inline contribution/minikube/start_minikube.sh:8]: is this a custom image? is this based on kernel 4.19.94?

Comment (h3llix) [reply contribution/minikube/start_minikube.sh:8]: Yes, its a custom image based on 4.19.94. We have a new repo named `kastore`

**Comment by nam-jaehyun** [comment]
Are we waiting for a new iso file with AppArmor?

**Comment by h3llix** [comment]
> Are we waiting for a new iso file with AppArmor?

Hi @nam-jaehyun. I have added AppAromor image on kastore.

**Pull request merged**